### PR TITLE
feat: Use protoc-gen-validate as gohub validator tool.

### PR DESCRIPTION
**Pull request by shuofan**
#### What this PR does / why we need it:
Use protoc-gen-validate as gohub validator tool

#### Which issue(s) this PR fixes:
Fixes #

#### Specified Reviewers:
/assign @sfwn

### CODE REVIEW

File: tools/gohub/cmd/tools/install/install.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ func IncludeDirs() []string {
 	return []string{
 		filepath.Join(home, "."+cmd.Name, repo, "tools/protoc/include"),
 		filepath.Join(home, "."+cmd.Name),
+		filepath.Join(home, "."+cmd.Name, "github.com/envoyproxy/protoc-gen-validate"),
```
Comment (sfwn) [inline tools/gohub/cmd/tools/install/install.go:45]: why we doesn't need include proto files inside github.com/mwitkow/go-proto-validators before?

File: tools/gohub/cmd/tools/install/install.go [hunk 2]
```diff
@@ -129,10 +130,15 @@ func Download(override, verbose bool) {
 			// build
 			fmt.Printf("building %s ...\n", p.Name)
 			buildDir := repodir
-			if len(p.Path) > 0 {
-				buildDir = filepath.Join(repodir, p.Path)
+
+			if p.Name == "protoc-gen-validate" {
+				runCommand(buildDir, []string{fmt.Sprintf("GOBIN=%s", dir)}, "make", "build")
```
Comment (sfwn) [inline tools/gohub/cmd/tools/install/install.go:135]: go build is also works for envoyproxy/protoc-gen-validate

**Review by sfwn — CHANGES_REQUESTED** [review]

File: tools/gohub/cmd/tools/install/install.go (hunk 1 above)
Comment (shuofan) [reply tools/gohub/cmd/tools/install/install.go]: remove the include.

File: tools/gohub/cmd/tools/install/install.go [hunk 3]
```diff
@@ -129,10 +132,26 @@ func Download(override, verbose bool) {
...
+				err = os.RemoveAll(filepath.Join(dir, "include"))
+				cmd.CheckError(err)
+			} else {
+				if len(p.Path) > 0 {
+					buildDir = filepath.Join(repodir, p.Path)
+				}
+				src := filepath.Join(dir, p.Name)
+				runCommand(buildDir, nil, "go", "build", "-o", src)
+				runCommand(buildDir, nil, "cp", src, filepath.Join(goPath, "bin"))
```
Comment (sfwn) [inline tools/gohub/cmd/tools/install/install.go:153]: why not specify target output path of go build directly in line 152?

**Review by sfwn — CHANGES_REQUESTED** [review]

File: tools/gohub/cmd/tools/install/install.go (hunk 3 above)
Comment (sfwn) [reply tools/gohub/cmd/tools/install/install.go:153]: put all things under ~/.gohub is better I think.

Comment (sfwn) [reply tools/gohub/cmd/tools/install/install.go:153]: put all things under ~/.gohub is better I think.

File: tools/gohub/cmd/tools/install/install.go [hunk 4]
```diff
@@ -129,10 +132,26 @@ func Download(override, verbose bool) {
 			// build
 			fmt.Printf("building %s ...\n", p.Name)
 			buildDir := repodir
-			if len(p.Path) > 0 {
-				buildDir = filepath.Join(repodir, p.Path)
+			goPath := os.Getenv("GOPATH")
+			if p.Name == "protoc-gen-validate" {
+				paths := []string{dir}
```
Comment (sfwn) [inline tools/gohub/cmd/tools/install/install.go:137]: it's better to wrap L137~L142 into a function variable and use it in L143

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**